src/main/java/server/HeadersParser.java

```diff
@@ -0,0 +1,39 @@
+package server;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class HeadersParser {
+    public Map<String, String> parse(ReadableWriteable readableWriteable) throws IOException {
+        String rawHeader = readableWriteable.readLine();
+        Map<String, String> headers = new java.util.HashMap<>(Map.of());
+
+        while (additionalHeadersRemain(rawHeader)) {
+            Header header = parseHeader(rawHeader);
+            headers.put(header.key, header.value);
+            rawHeader = readableWriteable.readLine();
+        }
+
+        return headers;
+    }
+
+    private boolean additionalHeadersRemain(String rawHeader) {
+        boolean isCRLF = rawHeader.equals(RequestParser.CRLF);
```

Comment:
**suggestion**: Move the `CRLF` in order to break a circular dependency.

As it stands right now, `RequestParser` depends on `HeadersParser`, and `HeadersParser` depends on `RequestParser`. 

```mermaid
flowchart
    RequestParser --> HeadersParser
    HeadersParser --> RequestParser
```